README.md

```diff
@@ -114,3 +114,142 @@ You could also do so using [Weinre](https://people.apache.org/~pmuellr/weinre/do
 The image was optimized to limit rebuilding and to be run in a developer-centric way. `scripts/rundocker.sh` describes the options needed to get the build running properly.
 
 Unless you need to make edits to the build method or are debugging one of the build dependencies and would like to continue using docker, you shouldn't need to modify that script.
+
+## Using the Android Emulator
+
+Installing the APK to a real device during development is slow. Instead, the
+[Android Emulator](https://developer.android.com/studio/run/emulator) can be
+used for faster installs in a clean environment. The recommended way to use
+the emulator is through Android Studio, but it can also be invoked directly.
+
+First, ensure the emulator is installed in the Android SDK. The `make setup`
+target will install the necessary pieces. Assuming the SDK is installed in
+`~/.android/sdk`:
+
+```
+~/.android/sdk/cmdline-tools/latest/bin/sdkmanager emulator
+```
+
+Next, install a platform and system image:
+
+```
+~/.android/sdk/cmdline-tools/latest/bin/sdkmanager "platforms;android-31"
+~/.android/sdk/cmdline-tools/latest/bin/sdkmanager "system-images;android-31;default;x86_64"
+```
+
+Now an [Android Virtual Device
+(AVD)](https://developer.android.com/studio/run/managing-avds) needs to be
+created. This can be done from the command line with
+[avdmanager](https://developer.android.com/studio/command-line/avdmanager).
+
+There are many device definitions available. To see the list, run:
+
+```
+~/.android/sdk/cmdline-tools/latest/bin/avdmanager list device
+```
+
+To create an AVD, a name, system image and device must be provided. Assuming
+the system image provided above and a Pixel 5 device:
+
+```
+~/.android/sdk/cmdline-tools/latest/bin/avdmanager create device --name test \
```

Comment:
In my case from toolbox this was `create avd` not `create device`.
```suggestion
~/.android/sdk/cmdline-tools/latest/bin/avdmanager create avd --name test \
```

Reply:
Oops, that is correct.